### ISSUE TITLE
Usernames are incorrectly sanitized 

### ISSUE THREAD

**Issue by jbweston** [jupyterhub/dockerspawner#205]
Dockerspawner provides the  `escaped_username` property, which is internally used to populate `username` field in the `container_name_template`.

If I understand correctly this is because docker containers may not start with an underscore. I note that when trying to start a docker container with the name `_`, docker informs me of the following:

```
docker: Error response from daemon: Invalid container name (_), only [a-zA-Z0-9][a-zA-Z0-9_.-] are allowed.
```

If  `container_name_template` is set to `{username}`, then we must ensure that usernames do indeed conform to the above regex. This does not, however, appear to be the case. 

Consider the following:

```
from types import SimpleNamespace
import dockerspawner

ds = dockerspawner.DockerSpawner()
ds.user = SimpleNamespace(name='_')  # mock
print(ds.escaped_name)  # prints '_5F' 

ds = dockerspawner.DockerSpawner()
ds.user = SimpleNamespace(name='hello_')  # mock
print(ds.escaped_name) # >>> 'hello_5F' 
```

It appears that `_` is being used as a marker for escaped characters (as `%` is in URLs) however, as previously noted, this is not safe.

**Comment by jfly** [comment]
See https://github.com/jupyterhub/dockerspawner/issues/326 for some more information on this. Currently, `_` *is* an allowed container name, but it is *not* allowed as the first character of a container name.

**Comment by minrk** [comment]
This will be closed by #414
